creator/lib/screens/login_screen.dart

```diff
@@ -8,10 +17,221 @@ class LoginScreen extends StatefulWidget {
+            'assets/signin_button/apple_logo.png',
+            'Appleでログイン',
+            () {
+              _tryLoginWith(
+                context,
+                signInWithApple(),
+              );
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildLoginButton(
+    BuildContext context,
+    String asset,
+    String text,
+    Function() method,
+  ) {
+    final mq = MediaQuery.of(context).size;
+    final screenWidth = mq.width;
+    final screenHeight = mq.height;
+
+    return Center(
+      child: Container(
+        width: screenWidth < 600 ? screenWidth * 0.8 : screenWidth * 0.5,
+        height: screenHeight * 0.06,
+        child: ElevatedButton(
+          style: Theme.of(context).elevatedButtonTheme.style?.copyWith(
+                backgroundColor: MaterialStateProperty.all(Colors.white),
+                foregroundColor: MaterialStateProperty.all(Colors.black),
+              ),
+          onPressed: method,
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              _buildLoginMethodLogo(asset),
+              SizedBox(
+                width: 16,
+              ),
+              Text(
+                text,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Container _buildLoginMethodLogo(String asset) {
+    final iconsSize = MediaQuery.of(context).size.width * 0.08;
+    return Container(
+      width: iconsSize,
+      height: iconsSize,
+      child: FittedBox(
+        fit: BoxFit.cover,
+        child: Image.asset(asset),
+      ),
+    );
+  }
+
+  // ログインできていたらHome、　出来ていなかったらsalon_applying_screen
+
+  //TODO: _addToDatabaseでデータベースに登録するが、現時点では実装していない。
+  void _navigateBasedOnRole() {
+    final _auth = FirebaseAuth.instance.currentUser;
+    print('宏輝: $_auth');
+    FirebaseFirestore.instance
+        .collection('users')
+        .doc(_auth?.email)
+        .get()
+        .then((DocumentSnapshot snapshot) {
+      if (snapshot.get(FieldPath(['role'])) == 0) {
+        Navigator.of(context).pushReplacementNamed('/salon_applying_screen');
+      } else if (snapshot.get(FieldPath(['role'])) == 1) {
+        FirebaseFirestore.instance
+            .collection('salons')
+            .where('owner', isEqualTo: _auth?.email)
+            .get()
+            .then((QuerySnapshot snapshot) {
+          print(snapshot.docs.isNotEmpty);
+          print(snapshot.docs.isEmpty);
+          if (snapshot.docs.isNotEmpty) {
+            Navigator.of(context)
+                .pushReplacementNamed(('/salon_applying_screen'));
+          } else {
+            Navigator.of(context)
+                .pushReplacementNamed(('/salon_applying_screen'));
+          }
+        });
+      }
+    });
+  }
+
+  Future<void> _addToDatabase() async {
+    DbHandler dbHandler = DbHandler();
+    final User? user = FirebaseAuth.instance.currentUser;
+
+    // データが無いとエラー可能性
+    if (user != null) {
```

Comment:
```suggestion
    if (user.email != null) {
```

If you do this, you don't have to do the email! on L147

Reply:
I changed.